src/instrumenter/annotations.ts

```diff
@@ -351,6 +359,28 @@ class AnnotationExtractor {
                     target
                 );
             }
+        } else if (
+            target instanceof Statement ||
+            target instanceof Block ||
+            target instanceof UncheckedBlock
+        ) {
+            if (annotation.type !== AnnotationType.Assert) {
+                throw new UnsupportedByTargetError(
+                    `The "${annotation.type}" annotation is not applicable inside functions`,
+                    annotation.original,
+                    annotation.annotationFileRange,
+                    target
+                );
+            }
+
+            if (target instanceof TryCatchClause) {
+                throw new UnsupportedByTargetError(
```

Comment:
Leaving as is for now, as it would require re-factoring `UnsupportedByTargetError` and `PPAbleError` to contain `OffsetRange`s instead of `Range`s and `unit` references, and convert those to `Range`s in `scribble.ts` similarly to how `STypeError` and `SemError` are handled